### ISSUE TITLE
[Bug]: Inlay-hint not working

### ISSUE THREAD

**Issue by malinthar** [ballerina-platform/ballerina-lang#40543]
### Description

$subject for the following source code

```ballerina
import ballerina/io;
import ballerina/http;
import ballerinax/openai.audio;
import ballerinax/openai.text;

configurable string openAPIToken = ?;

const string AUDIO_FILE = "german_hello.mp3";
const string AUDIO_FILE_PATH = "./audo_clips/" + AUDIO_FILE;

public function main(string audioURL, string toLanguage) returns error? {
    http:Client audioEP = check new (audioURL);
    http:Response httpResp = check audioEP->/get();
    byte[] audioBytes = check httpResp.getBinaryPayload();
    check io:fileWriteBytes(AUDIO_FILE_PATH, audioBytes);

    // Creates a request to translate the audio file to text (English)
    audio:CreateTranslationRequest translationReq = {
        file: {fileContent: check io:fileReadBytes(AUDIO_FILE_PATH), fileName: AUDIO_FILE},
        model: "whisper-1"
    };

    //Translate the audio file to text
    audio:Client openAIAudio = check new ({auth: {token: openAPIToken}});
    audio:CreateTranslationResponse translationRes = check openAIAudio->/audio/translations.post(translationReq);
    io:println("Audio text in English: ", translationRes);

    // Creates a request to translate the text from English to another language
    text:CreateCompletionRequest completionReq = {
        model: "text-davinci-003",
        prompt: string `Translate the following text from English to ${toLanguage}: ${translationRes.text}`,
        temperature: 0.7,
        max_tokens: 256,
        top_p: 1,
        frequency_penalty: 0,
        presence_penalty: 0
    };

    // Translates the text from English to another language
    text:Client openAIText = check new ({auth: {token: openAPIToken}});
    text:CreateCompletionResponse completionRes = check openAIText->/completions.post(completionReq);
    string? translatedText = completionRes.choices[0].text;

    if translatedText is () {
        return error("Failed to translate the given audio.");
    }
    io:println("Translated text: ", translatedText);

    var a = test("sampel", completionReq);
}

function test(string param1, text:CreateCompletionRequest param2) returns string {
    return "string";
}

```

### Affected Version(s)

2201.6.0-rc1

### Related area

-> Editor

**Comment by malinthar** [comment]
Sub task of #40576

**Comment by malinthar** [comment]
![image](https://github.com/ballerina-platform/ballerina-lang/assets/35211477/5852b00f-7061-4c80-8582-26f8294b54dc)

Working with the fix #40630
